### DISCUSSION TITLE
SDR++ server mode bi-directional VFO sync

### DISCUSSION THREAD

**Discussion by bturbeville** [AlexandreRouma/SDRPlusPlus#1370] [Q&A]
I have implemented a plugin that utilizes the ICOM 7610 transceiver's IQ data in SDR++.  The 7610 IQ USB port implements several ICOM CIV commands for controlling the transceiver including setting/getting VFO data.  When running my plugin standlone connected to the transceiver via USB bi-directional sync of the VFO in SDR++ and on the ICOM works.  Change the VFO on the transceiver and the VFO in SDR++ changes and vice versa.  Running in client/server mode with the server running on Ubuntu 22.04 and the client running on windows or linux works in terms of receiving/decoding the IQ data and syncing the transceivers VFO with SDR++ but changes to the transceiver VFO do not sync SDR++ client VFO.  Investigating the SDR++ client/server mode appears that VFO updates only go from client to server.  Does anyone have any ideas about implementing the bi-directional sync in client/server mode with the existing SDR++ server or do I need to just extend the network client and server to push the VFO changes back to the client when the server side VFO gets changed?

**Comment by AlexandreRouma** [comment]
idk what you're doing exactly but the server doesn't do anything with VFOs, it only deals with the center IQ frequency because that's all it sends back to the client...
